observer/holder/nodesHolder_test.go

```diff
@@ -0,0 +1,200 @@
+package holder
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-core-go/core/check"
+	"github.com/multiversx/mx-chain-proxy-go/data"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNodesHolder_ConstructorAndGetters(t *testing.T) {
+	nh, err := NewNodesHolder([]*data.NodeData{}, []*data.NodeData{}, data.AvailabilityAll)
+	require.Equal(t, errEmptyNodesList, err)
+	require.True(t, check.IfNil(nh))
+
+	syncedNodes := createTestNodes(3)
+	setPropertyToNodes(syncedNodes, "synced", true, 0, 1, 2)
+
+	fallbackNodes := createTestNodes(3)
+	setPropertyToNodes(fallbackNodes, "synced", true, 0, 1, 2)
+	setPropertyToNodes(fallbackNodes, "fallback", true, 0, 1, 2)
+
+	nh, err = NewNodesHolder(syncedNodes, fallbackNodes, data.AvailabilityAll)
+	require.NoError(t, err)
+	require.False(t, nh.IsInterfaceNil())
+
+	require.Equal(t, []*data.NodeData{syncedNodes[0]}, nh.GetSyncedNodes(0))
+	require.Equal(t, []*data.NodeData{syncedNodes[1]}, nh.GetSyncedNodes(1))
+	require.Equal(t, []*data.NodeData{syncedNodes[2]}, nh.GetSyncedNodes(core.MetachainShardId))
+
+	require.Equal(t, []*data.NodeData{fallbackNodes[0]}, nh.GetSyncedFallbackNodes(0))
+	require.Equal(t, []*data.NodeData{fallbackNodes[1]}, nh.GetSyncedFallbackNodes(1))
+	require.Equal(t, []*data.NodeData{fallbackNodes[2]}, nh.GetSyncedFallbackNodes(core.MetachainShardId))
+
+	setPropertyToNodes(syncedNodes, "synced", false, 0, 2)
+	setPropertyToNodes(fallbackNodes, "synced", false, 1)
+	nh.UpdateNodes(append(syncedNodes, fallbackNodes...))
+	require.Equal(t, []*data.NodeData{syncedNodes[0]}, nh.GetOutOfSyncNodes(0))
+	require.Equal(t, []*data.NodeData{}, nh.GetOutOfSyncNodes(1))
+	require.Equal(t, []*data.NodeData{syncedNodes[2]}, nh.GetOutOfSyncNodes(core.MetachainShardId))
+
+	require.Equal(t, []*data.NodeData{}, nh.GetOutOfSyncFallbackNodes(0))
+	require.Equal(t, []*data.NodeData{fallbackNodes[1]}, nh.GetOutOfSyncFallbackNodes(1))
+	require.Equal(t, []*data.NodeData{}, nh.GetOutOfSyncFallbackNodes(core.MetachainShardId))
```

Comment:
so much for "a test should test one thing"

Reply:
refactored